lib/cmd/common/build.js

```diff
@@ -119,29 +120,61 @@ function argsToPayload(args) {
 }
 
 /**
- * Mutates the arguments object in order to support --branch --commit, turning them into gitRef
+ * Mutates the arguments object in order to support --git-branch --git-commit and --git-tag, turning them into gitRef
  * @param  {Object} args Arguments object
  */
 function mutateGitRefArgs(args) {
-  if (args.branch) {
+  // Workaround for current millicore behavior:
+  // If no git- params are present, build from branch:master
+  var gitParams = [
+    'git-commit',
+    'git-tag',
+    'git-branch'
+  ];
+  if (!_.any(gitParams, _.propertyOf(args))) {
     args.gitRef = {
       type: 'branch',
-      value: args.branch
+      value: 'master'
     };
+    return;
   }
 
-  if (args.commit) {
-    if (!args.branch) {
-      throw new Error("commit should be used along with a corresponding branch!");
-    }
-    args.gitRef.hash = args.commit;
+  function clearGitParams(args) {
+    _.each(gitParams, function(p) {
+      delete args[p];
```

Comment:
It's just not to send extra parameters that the back-end doesn't know about. It's optional but I just think not sending extra stuff in the request is better.